Microsoft.Azure.Cosmos.Samples/Usage/CustomDiagnosticAndEventListener/CustomDiagnosticAndEventListener.cs

```
@@ -0,0 +1,167 @@
+﻿namespace Sample.Listeners
+{
+    using System.Diagnostics.Tracing;
+    using System.Diagnostics;
+    using System.Collections.Concurrent;
+
+    internal class CustomDiagnosticAndEventListener :
+        EventListener, // Override Event Listener to capture Event source events
+        IObserver<KeyValuePair<string, object>>, // Override IObserver to capture Activity events
+        IObserver<DiagnosticListener>,
+        IDisposable
+    {
+        private readonly string diagnosticSourceName;
+        private readonly string eventSourceName;
+
+        private ConcurrentBag<IDisposable>? Subscriptions = new();
+        private ConcurrentBag<Activity> Activities { get; } = new();
+
+        public CustomDiagnosticAndEventListener(string diagnosticSourceName, string eventSourceName)
+        {
+            this.diagnosticSourceName = diagnosticSourceName;
+            this.eventSourceName = eventSourceName;
+
+            DiagnosticListener.AllListeners.Subscribe(this);
+        }
+
+        /// <summary>
+        /// IObserver Override
+        /// </summary>
+        public void OnCompleted() {
+            throw new NotImplementedException(); 
+        }
+
+        /// <summary>
+        /// IObserver Override
+        /// </summary>
+        public void OnError(Exception error) {
+            throw new NotImplementedException(); 
```

Can we throw the actual exception instead of a NotImplementedException?

---

Logging the exception